src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -196,14 +196,19 @@ internal static AuthorityType GetAuthorityType(string authority)
             }
         }
 
-        internal abstract string GetTenantId();
+        internal abstract string TenantId { get; }
 
         /// <summary>
         /// Gets a tenanted authority if the current authority is tenantless.
         /// Returns the original authority on B2C and ADFS
         /// </summary>
         internal abstract string GetTenantedAuthority(string tenantId);
 
+        /// <summary>
+        /// Gets the authority endpoints based on the host and tenant id. Does not rely on OIDC discovery.
+        /// </summary>
+        internal abstract AuthorityEndpoints GetHardcodedEndpoints();
```

Comment:
nit: just for discussion...these are actually the authorization server endpoints...Does it make sense to call them what they are, verse authority endpoints? just seems confusing. really just a discussion point here.

Reply:
Happy to rename them, but maybe not Authorization endpoints because "authoirzation endpoint" is one of them. Myabe OAuth2Endpoints?

Reply:
OAuth2Endpoints make sense.